src/main/java/org/apache/flink/checkpoint/benchmark/FsStateBackendRescalingBenchmarkExecutor.java

```diff
@@ -0,0 +1,89 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.checkpoint.benchmark;
+
+import org.apache.flink.api.common.JobID;
+import org.apache.flink.config.ConfigUtil;
+import org.apache.flink.config.StateBenchmarkOptions;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.contrib.streaming.state.benchmark.RescalingBenchmarkBuilder;
+import org.apache.flink.runtime.state.hashmap.HashMapStateBackend;
+import org.apache.flink.runtime.state.storage.FileSystemCheckpointStorage;
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.runner.RunnerException;
+
+import java.io.IOException;
+import java.net.URI;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static org.openjdk.jmh.annotations.Mode.AverageTime;
+
+@OutputTimeUnit(MILLISECONDS)
+@BenchmarkMode(AverageTime)
+@Warmup(iterations = 3)
+public class FsStateBackendRescalingBenchmarkExecutor extends RescalingBenchmarkBase {
+    // numberOfKeys = 1250000, keyLen = 96, valueLen = 128, state size ~= 270MB
+    private final int numberOfKeys = 1250000;
+    private final int keyLen = 96;
+
+    public static void main(String[] args) throws RunnerException {
+        runBenchmark(RocksdbStateBackendRescalingBenchmarkExecutor.class);
+    }
+
+    @Setup(Level.Trial)
+    public void setUp() throws Exception {
+        // FsStateBackend is deprecated in favor of HashMapStateBackend with setting checkpointStorage.
+        HashMapStateBackend stateBackend = new HashMapStateBackend();
+        Configuration benchMarkConfig = ConfigUtil.loadBenchMarkConf();
+        String stateDataDirPath = benchMarkConfig.getString(StateBenchmarkOptions.STATE_DATA_DIR);
+        benchmark =
+                new RescalingBenchmarkBuilder<byte[]>()
+                        .setMaxParallelism(128)
+                        .setParallelismBefore(rescaleType.getParallelismBefore())
+                        .setParallelismAfter(rescaleType.getParallelismAfter())
+                        .setCheckpointStorageAccess(
+                                new FileSystemCheckpointStorage(new URI("file://" + stateDataDirPath), 0)
+                                        .createCheckpointStorage(new JobID()))
+                        .setStateBackend(stateBackend)
+                        .setStreamRecordGenerator(new ByteArrayRecordGenerator(numberOfKeys, keyLen))
+                        .setStateProcessFunctionSupplier(TestKeyedFunction::new)
+                        .build();
+        benchmark.setUp();
+    }
+
+    @Setup(Level.Invocation)
+    public void setUpPerInvocation() throws Exception {
+        benchmark.prepareStateForOperator(rescaleType.getSubtaskIndex());
+    }
```

Comment:
Do we need to use `Level.Invocation`? Invocation level setup/tear downs are problematic:

http://javadox.com/org.openjdk.jmh/jmh-core/1.7/org/openjdk/jmh/annotations/Level.html#Invocation

Reply:
Thanks for your comments, changed `Level.Invocation` to `Level.Iteration`.  After changing, the results are about the same as before.

Reply:
That's good to hear :)